src/include/daos_pipeline.h

```diff
@@ -0,0 +1,281 @@
+/**
+ * (C) Copyright 2021 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+
+#ifndef __DAOS_PIPE_H__
+#define __DAOS_PIPE_H__
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <daos_types.h>
+#include <daos_obj.h>
+
+
+/**
+ * A filter part object, used to build a filter object for a pipeline.
+ *
+ */
+typedef struct {
+	/**
+	 *  Part type can be any of the following:
+	 *   -- function:
+	 *      - logical functions:
+	 *          DAOS_FILTER_FUNC_EQ:		==
+	 *          DAOS_FILTER_FUNC_NE:		!=
+	 *          DAOS_FILTER_FUNC_LT:		<
+	 *          DAOS_FILTER_FUNC_LE:		<=
+	 *          DAOS_FILTER_FUNC_GE:		>=
+	 *          DAOS_FILTER_FUNC_GT:		>
+	 *          DAOS_FILTER_FUNC_IN:		IN (const1, const2, ...)
+	 *          DAOS_FILTER_FUNC_LIKE:		== (reg exp.)
+	 *          DAOS_FILTER_FUNC_ISNULL:		==NULL
+	 *          DAOS_FILTER_FUNC_ISNOTNULL:		!=NULL
+	 *          DAOS_FILTER_FUNC_AND:		&&
+	 *          DAOS_FILTER_FUNC_OR:		||
+	 *          DAOS_FILTER_FUNC_NOT:		!
+	 *      - artihmetic functions:
+	 *          DAOS_FILTER_FUNC_ADD:		+
+	 *          DAOS_FILTER_FUNC_SUB:		-
+	 *          DAOS_FILTER_FUNC_MUL:		*
+	 *          DAOS_FILTER_FUNC_DIV:		/
+	 *      - aggeration functions:
```

Comment:
```suggestion
	 *      - aggregation functions:
```

Reply:
Thanks @nzrutman, I will push this change to feature/pipeline_api